__main__.py

```diff
@@ -26,7 +26,7 @@ def main():
         if not args.json:
             print(exc, file=sys.stderr)
         else:
-            json.dump(exc.asdict(), sys.stdout)
+            json.dump([e.asdict() for e in getattr(exc, "errors", [exc])], sys.stdout, indent=2)
```

Comment:
Can we actually catch both cases separately CollectedValidationErrors, ValidationError instead, or refactor ValidationError to always point to a list of errors.